controllers/keda/hpa.go

```diff
@@ -166,15 +166,17 @@ func (r *ScaledObjectReconciler) getScaledObjectMetricSpecs(logger logr.Logger,
 		return nil, err
 	}
 
-	for _, scaler := range scalers {
+	for scalerIndex, scaler := range scalers {
 		metricSpecs := scaler.GetMetricSpecForScaling()
-
 		for _, metricSpec := range metricSpecs {
 			if metricSpec.Resource != nil {
 				resourceMetricNames = append(resourceMetricNames, string(metricSpec.Resource.Name))
 			}
 
 			if metricSpec.External != nil {
+				// Add the scaler index prefix
+				metricSpec.External.Metric.Name = fmt.Sprintf("s%d-%s", scalerIndex, metricSpec.External.Metric.Name)
```

Comment:
I don't like that we are adding the prefix here, imho `scaler.GetMetricSpecForScaling()` should return the final metric name.